### PR TITLE
chore: Examples pruning

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Description

Move pruning-marked examples to a legacy directory. Remove their download links and references from docs. Remove their tests from CI. 

Remove tutorials and model hub v1 examples docs.

## Test Plan

CI passes

## Commentary (optional)

Legacy folder should be migrated to a new repository in a follow-up PR.

## Checklist

- [x] Changes have been manually QA'd
- [x] Docs and github references to the legacy examples have been resolved
- [ ] Blog post references to the legacy examples have been resolved
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
MLG-869

### CODE REVIEW

**Review by tara-det-ai — APPROVED** [review]
tested that the docs pages are rendering as expected

File: docs/example-solutions/examples.rst [hunk 1]
```diff
@@ -19,7 +19,8 @@ For example, here is how to train the ``mnist_pytorch`` example with a fixed set
    cd mnist_pytorch
    det experiment create const.yaml .
 
-For an introduction to using the training API, please visit the Training APIs section.
+For an introduction to using the training APIs, please visit :ref:`Training APIs
+<apis-howto-overview>`.
```
[the remark this answers is not in the document]
Comment (MikhailKardash) [reply docs/example-solutions/examples.rst:23]: This link was never there to begin with, see: https://github.com/determined-ai/determined/pull/5791/files#r1088118342

File: examples/Makefile [hunk 2]
```diff
@@ -90,15 +66,7 @@ build/%.tgz: */%/ build/newest build/manifest
...
 
-test:
-	pytest -vv -s --durations=0 tests
-
 fmt:
 	make -C tests fmt
 	black .
 	isort .
-
-check:
```
Comment (azhou-determined) [inline examples/Makefile:101]: is it worth keeping these build steps for the examples that are remaining in `examples`? i get that we don't have tests for them, but maybe we should add them in the future?

**Comment by azhou-determined** [comment]
not necessarily in scope for this PR, but maybe we should reorganize the "un-pruned" examples in the `examples` directory. like with these removals, some of the directories don't make as much sense anymore. 

we have a bunch that are parent directories with nested single child directories:
`computer_vision/cifar10_tf_keras`
`deepspeed/gpt_neox`
`deepspeed_autotune/torchvision`
`diffusion/textual_inversion_stable_diffusion`

might require more thought, but there's probably a better way to organize them. maybe just by platform? `deepspeed`, `pytorch`, `tf_keras`, etc.?

File: examples/Makefile (hunk 2 above)
Comment (MikhailKardash) [reply examples/Makefile:101]: On second thought, it's definitely a mistake to remove `check`, since it checks formatting of every example.

I do, however, think that `examples/tests` is redundant. We use examples in either `e2e_tests/test` or `harness/tests` depending on if it's a unit/integration/capability test. Any tests that are actively maintained in `determined` should be serving CI or serving Docs, as a showcase of an important feature that we maintain.

**Review by azhou-determined — APPROVED** [review]
nice!

**Comment by MikhailKardash** [comment]
Github won't let me merge and insists on quite a large rebase after some force pushing. Closing this and making a new one. Sorry all. :|
